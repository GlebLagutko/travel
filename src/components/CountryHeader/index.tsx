import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import {createStyles, makeStyles, Theme} from '@material-ui/core/styles';
import {SelectLanguage} from "../SelectLanguage";
import {Link} from 'react-router-dom';
import {CHANGE_COUNTRY} from "../../store/actions/ChangeCountry";
import {useDispatch, useSelector} from "react-redux";
import {Home} from "@material-ui/icons";
import {UserInfo} from "../UserInfo";
import {saveState} from "../../store/SaveState";
import store from "../../store";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            flexGrow: 1,
            width: "100%",
        },

        title: {
            flexGrow: 1,
            display: 'none',
            [theme.breakpoints.up('sm')]: {
                display: 'block',
            },
        },
        home: {
            display: 'flex',
            cursor: 'pointer',
            [theme.breakpoints.up('sm')]: {
                display: 'block',
            },

        },

    }),
);

interface HeaderProps {
    changeHolder(newValue: string): void
}

const countryState = state => state.value.country;

export default function CountryHeader({country}) {
    const classes = useStyles();

    const dispatch = useDispatch();
    const countryInfo = useSelector(countryState);

    return (
        <div className={classes.root}>
            <AppBar position="static">
                <Toolbar>
                    <Typography onClick={() => {
                        dispatch({type: CHANGE_COUNTRY, value: {urlName: 'Main page'}});
                        saveState(store);
                    }} className={classes.home + ' home-icon'} variant="h6" noWrap>
                        <Home/>
                    </Typography>
                    <Typography className={classes.title} variant="h6" noWrap>
                        {country}
                    </Typography>
                    <SelectLanguage/>
                    <UserInfo/>
                </Toolbar>
            </AppBar>
        </div>
    );
}
