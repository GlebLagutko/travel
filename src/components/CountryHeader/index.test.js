import {act, create} from 'react-test-renderer';

import React from "react";
import CountryHeader from ".";
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'

describe('CountryHeader', () => {
  it('renders correctly', async () => {

    const initialState = {value: {language: "en"}};
    const mockStore = configureStore();
    let store = mockStore(initialState);

    const testRenderer = create(
      <Provider store={store}><CountryHeader/></Provider>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();

  });

});