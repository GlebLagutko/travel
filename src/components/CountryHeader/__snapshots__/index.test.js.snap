// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryHeader renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-home-3 home-icon MuiTypography-h6 MuiTypography-noWrap"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
      </h6>
      <h6
        className="MuiTypography-root makeStyles-title-2 MuiTypography-h6 MuiTypography-noWrap"
      />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.87 15.07l-2.54-2.51.03-.03c1.74-1.94 2.98-4.17 3.71-6.53H17V4h-7V2H8v2H1v1.99h11.17C11.5 7.92 10.44 9.75 9 11.35 8.07 10.32 7.3 9.19 6.69 8h-2c.73 1.63 1.73 3.17 2.98 4.56l-5.09 5.02L4 19l5-5 3.11 3.11.76-2.04zM18.5 10h-2L12 22h2l1.12-3h4.75L21 22h2l-4.5-12zm-2.62 7l1.62-4.33L19.12 17h-3.24z"
          />
        </svg>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-label="Without label"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            En
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value="en"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        className="login"
        onClick={[Function]}
      >
        Login
      </div>
    </div>
  </header>
</div>
`;
